src/main/java/com/flab/buywithme/service/common/CommonPostService.java

```diff
@@ -0,0 +1,35 @@
+package com.flab.buywithme.service.common;
+
+import com.flab.buywithme.domain.Post;
+import com.flab.buywithme.domain.enums.PostStatus;
+import com.flab.buywithme.error.CustomException;
+import com.flab.buywithme.error.ErrorCode;
+import com.flab.buywithme.repository.PostRepository;
+import javax.persistence.LockModeType;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class CommonPostService {
+
+    private final PostRepository postRepository;
+
+    public Post getPost(Long postId) {
+        return postRepository.findById(postId)
+                .orElseThrow(() -> new CustomException(ErrorCode.POST_NOT_FOUND));
+    }
+
+    @Lock(value = LockModeType.PESSIMISTIC_WRITE)
```

Comment:
Lock annotation 에 대한 java doc 을 한번 읽어 보시고 service layer 에서 사용할수있는지 확인해주세요

Reply:
> Annotation used to specify the LockModeType to be used when executing the query. It will be evaluated when using Query on a query method or if you derive the query from the method name.
-> repository layer에서 사용하도록 수정했습니다.